File: src/main/kotlin/com/nuutrai/reactor/listeners/PlayerDeath.java
package com.nuutrai.reactor.listeners;

import com.nuutrai.reactor.data.DataManager;
import com.nuutrai.reactor.store.Store;
import org.bukkit.Bukkit;
import org.bukkit.Location;
import org.bukkit.World;
import org.bukkit.entity.Player;
import org.bukkit.event.EventHandler;
import org.bukkit.event.Listener;
import org.bukkit.event.entity.PlayerDeathEvent;

import static com.nuutrai.reactor.Reactor.instance;

public class PlayerDeath implements Listener {
    @EventHandler
    public void onPlayerDeath(PlayerDeathEvent e) {
        e.setDroppedExp(0);

        DataManager.loadPlayerData(e.getPlayer());

        Player p = e.getPlayer();

        Store.setup(p);

        World world = DataManager.get(p).getPlayer().getWorld();
        Bukkit.getScheduler().runTask(instance, () -> {
            Bukkit.getScheduler().runTaskLater(instance, () -> {
                p.teleport(new Location(world, 0, 121, 0));
                p.setAllowFlight(true);
            }, 20);
        });
    }
}

File: src/main/kotlin/com/nuutrai/reactor/listeners/PlayerDeath.kt
package com.nuutrai.reactor.listeners

import com.nuutrai.reactor.Reactor
import com.nuutrai.reactor.data.DataManager
import com.nuutrai.reactor.store.Store.setup
import org.bukkit.Bukkit
import org.bukkit.Location
import org.bukkit.event.EventHandler
import org.bukkit.event.Listener
import org.bukkit.event.entity.PlayerDeathEvent

class PlayerDeath : Listener {
	@EventHandler
	fun onPlayerDeath(e: PlayerDeathEvent) {
		e.droppedExp = 0

		DataManager.loadPlayerData(e.player)

		val p = e.player

		setup(p)

		val world = DataManager.get(p)?.getPlayer()?.world ?: return
		Bukkit.getScheduler().runTask(Reactor.Companion.instance!!, Runnable {
			Bukkit.getScheduler().runTaskLater(Reactor.Companion.instance!!, Runnable {
				p.teleport(Location(world, 0.0, 121.0, 0.0))
				p.allowFlight = true
			}, 20)
		})
	}
}
